tests/scripts/all.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+
+import subprocess
+import sys
+from types import FunctionType
+
+BASH_COMPONENT_CODE = '''
+global component_{0}
+def component_{0}():
+    component_proc = subprocess.Popen(['tests/scripts/all.sh', '{0}'], stdout=subprocess.PIPE)
+    for line in component_proc.stdout:
+        line = str(line, 'utf-8')
+        print(line, end='')
+'''
+
+def import_bash_components_file(components_file):
```

Comment:
Please use type annotations systematically (i.e. every function has a `-> …` annotation) in new Python code. They make it a lot easier to understand what the code does.